refract-spec.md

```diff
@@ -107,34 +89,511 @@ In addition to expressing Refract Elements as objects with `element`, `attribute
+
+#### Properties
+- `element`: number (string, fixed)
+- `content` (number, required)
+
+#### Examples
+
+In JSON, a number is represented as:
+
+```json
+400
+```
+
+In Refract, this number is expanded to a `Number Element`.
+
+```json
+{
+  "element": "number",
+  "attributes": {},
+  "content": 400
+}
+```
+### Boolean Element (Element)
+
+A `Boolean Element` allows for values that are JSON boolean.
+
+#### Properties
+- `element`: boolean (string, fixed)
+- `content` (boolean, required)
+
+#### Examples
+
+In JSON, an example of a boolean is:
+
+```json
+true
+```
+
+In Refract, this boolean is expanded to a `Boolean Element`.
+
+```json
+{
+  "element": "boolean",
+  "attributes": {},
+  "content": true
+}
+```
+
+### Array Element (Element)
+
+An `Array Element` allows for JSON arrays.
+
+#### Properties
+- `element`: array (string, fixed)
+- `content` (array[Element], required)
+
+
+#### Examples
+
+In JSON, an example of an array is:
+
+```json
+["foo", 400, true]
+```
+
+In Refract, this boolean is expanded to a `Boolean Element`.
+
+```json
+{
+  "element": "array",
+  "attributes": {},
+  "content": [
+    {
+      "element": "string",
+      "attributes": {},
+      "content": "foo"
+    },
+    {
+      "element": "number",
+      "attributes": {},
+      "content": 400
+    },
+    {
+      "element": "boolean",
+      "attributes": {},
+      "content": true
+    }
+  ]
+}
+```
+
+### Property Element (Element)
+
+A `Property Element` allows for values that MUST be used in `Object Element` elements.
+
+#### Properties
+- `element`: property (string, fixed)
+- `attributes`
+    - `name` (string, required) - Name of property
+- `content` (Element, required) - Any element that extends Element
+
+#### Examples
+
+To see how this element is used, view the `Object Element` definition.
+
+### Object Element (Element)
+
+A `Object Element` allows for values that are JSON boolean.
+
+#### Properties
+- `element`: object (string, fixed)
+- `content` (array[Property Element], required)
+
+#### Examples
+
+In JSON, an example of an object is:
+
+```json
+{
+  "foo": "bar"
+}
+```
+
+In Refract, this object is expanded to a `Object Element`.
+
+```json
+{
+  "element": "object",
+  "attributes": {},
+  "content": [
+    {
+      "element": "property",
+      "attributes": {
+        "name": "foo"
+      },
+      "content": {
+        "element": "string",
+        "attributes": {},
+        "content": "bar"
+      }
+    }
+  ]
+}
+```
+
+## Referencing Elements
+
+The referencing elements are provided as part of the base specification for the purpose of identifying and referencing elements.
+
+### ID (Element)
+
+The `id` element is a way to identify an element in a document. IDs MUST be unique throughout a document.
+
+#### Properties
+
+- `element` id (string, fixed)
+- `content` (string) - Unique identifier
+
+#### Example
+
+```json
+{
+  "element": "string",
+  "attributes": {
+    "id": "foo"
+  },
   "content": "bar"
 }
 ```
 
```

Comment:
@fosrias Yes, we decided to take it out. Do you all think it is necessary? If you have actually altered the document to replace the ref element with the element in which it was referencing, is it really transcluding anymore? Isn't it a new element at that point?